vite.config.ts

```diff
@@ -36,25 +35,5 @@ export default defineConfig(({ ssrBuild }) => ({
     }),
     react(),
     tsconfigPaths(),
-    injectExternalReactWithDom(),
-    viteExternalsPlugin({ react: 'React', 'react-dom': 'ReactDOM' }),
   ],
 }));
-
-function injectExternalReactWithDom() {
```

Comment:
This code injects production version of react-dom (in dev mode too) -> react-refresh can't apply changes